packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,742 @@
+  }
+
+  private static buildJobArn(scope: constructs.Construct, jobName: string) : string {
+    return cdk.Stack.of(scope).formatArn({
+      service: 'glue',
+      resource: 'job',
+      resourceName: jobName,
+    });
+  }
+
+  /**
+   * The ARN of the job.
+   */
+  public readonly jobArn: string;
+
+  /**
+   * The name of the job.
+   */
+  public readonly jobName: string;
+
+  /**
+   * The IAM role associated with this job.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * Used to cache results of ._rule calls.
+   * @private
+   */
+  private _rules: Record<string, events.Rule> = {};
+
+  constructor(scope: constructs.Construct, id: string, props: JobProps) {
+    super(scope, id, {
+      physicalName: props.jobName,
+    });
+
+    // Create a basic service role if one is not provided https://docs.aws.amazon.com/glue/latest/dg/create-service-policy.html
+    this.role = props.role || new iam.Role(this, 'ServiceRole', {
+      assumedBy: new iam.ServicePrincipal('glue'),
+      // arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole
+      managedPolicies: [iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSGlueServiceRole')],
+    });
+
+    const jobResource = new CfnJob(this, 'Resource', {
+      name: props.jobName,
+      description: props.description,
+      role: this.role.roleArn,
+      command: {
+        name: props.jobCommand.name.name,
+        scriptLocation: props.jobCommand.scriptLocation,
+        pythonVersion: props.jobCommand.pythonVersion,
+      },
+      glueVersion: props.glueVersion ? props.glueVersion.name : undefined,
+      workerType: props.workerType ? props.workerType.name : undefined,
+      numberOfWorkers: props.numberOfWorkers,
+      maxCapacity: props.maxCapacity,
+      maxRetries: props.maxRetries,
+      executionProperty: props.maxConcurrentRuns ? { maxConcurrentRuns: props.maxConcurrentRuns } : undefined,
+      notificationProperty: props.notifyDelayAfter ? { notifyDelayAfter: props.notifyDelayAfter.toMinutes() } : undefined,
+      timeout: props.timeout ? props.timeout.toMinutes() : undefined,
+      connections: props.connections ? { connections: props.connections.map((connection) => connection.connectionName) } : undefined,
+      securityConfiguration: props.securityConfiguration ? props.securityConfiguration.securityConfigurationName : undefined,
+      defaultArguments: props.defaultArguments,
+      tags: props.tags,
+    });
+
+    const resourceName = this.getResourceNameAttribute(jobResource.ref);
+    this.jobArn = Job.buildJobArn(this, resourceName);
+    this.jobName = resourceName;
+  }
+
+
+  /**
+   * Create a CloudWatch Event Rule matching transition into the given `JobEventState`s
+   *
+   * @param state used in matching the CloudWatch Event
+   *
+   * @see https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/EventTypes.html#glue-event-types
+   */
+  public rule(state: JobEventState, props?: events.RuleProps): events.Rule {
```

Comment:
Looking for suggestions here... the `onXxx()` named methods have requirements around method signature that it should e.g. `error: [awslint:events-method-signature:@aws-cdk/aws-glue.Job.onStateChange.params[0]] all 'onXxx()' methods should have the CloudWatch Events signature (id: string, options: events.OnEventOptions = {})`.

The methods I implemented, did not allow the user to specify the construct id, as it internally set it and used it for caching the rules. The reason for caching the rules was so that the metrics methods don't end up creating new rules everytime they're called i.e. a rule is created once, metrics calls uses it.

The requirement to be able to specify the construct id for the rule, now really complicates this. The linting rules can be bypassed, but I wonder if that's the right thing to do here, or are there other patterns that can be used here?

Reply:
I've went with `onEvent` and followed the CloudWatch events method signatures as suggested by the linter